### ISSUE TITLE
Avatar fetching causes twofactor manager error to be logged

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#25290]
### Steps

To be confirmed.
I updated from 8.0 up to master, so it could be a side-effect of having older avatars

This appears when I reload the files app:
```
{"reqId":"ycGQMBfVJmETiDeIoOck","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: two-factor-providers at \/srv\/www\/htdocs\/owncloud\/lib\/private\/Authentication\/TwoFactorAuth\/Manager.php#111","level":0,"time":"2016-06-28T12:56:05+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/admin\/28","user":"admin"}
{"reqId":"ycGQMBfVJmETiDeIoOck","remoteAddr":"127.0.0.1","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/htdocs\/owncloud\/lib\/private\/Authentication\/TwoFactorAuth\/Manager.php#112","level":0,"time":"2016-06-28T12:56:05+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/admin\/28","user":"admin"}
{"reqId":"ycGQMBfVJmETiDeIoOck","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined index: two-factor-providers at \/srv\/www\/htdocs\/owncloud\/lib\/private\/Authentication\/TwoFactorAuth\/Manager.php#111","level":0,"time":"2016-06-28T12:56:05+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/admin\/28","user":"admin"}
{"reqId":"ycGQMBfVJmETiDeIoOck","remoteAddr":"127.0.0.1","app":"PHP","message":"Invalid argument supplied for foreach() at \/srv\/www\/htdocs\/owncloud\/lib\/private\/Authentication\/TwoFactorAuth\/Manager.php#112","level":0,"time":"2016-06-28T12:56:05+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/admin\/28","user":"admin"}
```

This is the stack I could extract with a debugger:
```
0  OC\Authentication\TwoFactorAuth\Manager->getProviders() /srv/www/htdocs/owncloud/lib/private/Authentication/TwoFactorAuth/Manager.php:111
1  OC\Authentication\TwoFactorAuth\Manager->isTwoFactorAuthenticated() /srv/www/htdocs/owncloud/lib/private/Authentication/TwoFactorAuth/Manager.php:66
2  OC\User\Session->isTwoFactorEnforced() /srv/www/htdocs/owncloud/lib/private/User/Session.php:358
3  OC\User\Session->logClientIn() /srv/www/htdocs/owncloud/lib/private/User/Session.php:310
4  OC\User\Session->tryBasicAuthLogin() /srv/www/htdocs/owncloud/lib/private/User/Session.php:397
5  OC::handleLogin() /srv/www/htdocs/owncloud/lib/base.php:927
6  OC::handleRequest() /srv/www/htdocs/owncloud/lib/base.php:867
7  {main}          /srv/www/htdocs/owncloud/index.php:39
```

This also triggers it:
```
% curl -X GET http://admin:admin@localhost/owncloud/index.php/avatar/admin/28 
```

I noticed that it goes there 4-5 times with correct values until it gets in this situation where `$info` only contains $info['version'] with the value 0.

@ChristophWurst

**Comment by PVince81** [comment]
Hmm, it doesn't happen when setting up master. Well, it happened only once when doing curl after setup, then not any more.

I see it's retrieving app infos. Maybe when I upgraded from 8.0 there were stray app entries in "oc_appconfig" for which there is no more info (apps that were bundled previously but where the code is missing).

Anyway, best would be to add a `isset` here just in case.

**Comment by PVince81** [comment]
Okay, got it:

1. Setup master
1. Clone the files_texteditor into the apps folder
1. Enable the text editor
1. Remove the "files_texteditor" folder (but the app is still enabled)
1. `curl -X GET http://admin:admin@localhost/owncloud/index.php/avatar/admin/28`

The message will appear every time for the `curl` command.

Ideally the apps API shouldn't even bother returning missing apps...

**Comment by ChristophWurst** [comment]
I guess the problem is that reading the info from app's info.xml usually gives you an array that has an index 'two-factor-providers', which is an empty array by default. Not sure if this is to be fixed in the 2fa code. We should probably not list missing apps when querying the installed ones.

**Comment by PVince81** [comment]
Looks like this is fixed in 9.1.1RC1.

**Comment by ChristophWurst** [comment]
was fixed by https://github.com/owncloud/core/pull/25606 :rocket:
